lib/cloudant.js

```diff
@@ -697,13 +705,15 @@ Cloudant.prototype.selectModel = function(model, migrate) {
     }
   }
 
+  var idName = this.idName(model);
   debug('Cloudant.prototype.selectModel use %j', dbName);
   this.pool[model] = {
     mo: mo,
+    dateFields: dateFields,
     db: this.cloudant.use(dbName),
+    idName: idName,
```

Comment:
@ssh24 please check the new function defined in https://github.com/strongloop/loopback-connector-cloudant/pull/158/files#diff-fa06ecf3249c63aabcbf7ff1ae8ec186R1199, hope it explains why we need it.